filters/all.js

```diff
@@ -531,6 +539,29 @@ function getAdditionalSubs(asyncapi, params) {
   return ret;
 }
 
+function getBrokerSettings(asyncapi,params){
+ let ret;
+
+ if(params.useServers === 'true'){
+    brokers = "";
+     for ( server in asyncapi.servers() ){
+       let url = ""
+       if (server.variable.port)) {
+         url = server.url();
+         url = url.replace('{port}', server.port.default);
```

Comment:
you expect that someone uses `port` word to specify port + that user always provides a default. This if fine, but needs to be documented in the readme.

Reply:
So I was following the example from here: https://raw.githubusercontent.com/asyncapi/asyncapi/2.0.0/examples/2.0.0/streetlights.yml
and I also saw that the a logic was being used in the nodejs project. at this line : https://github.com/asyncapi/nodejs-template/blob/fff91aa1f76db4682332f16bb91cdd78fb536ceb/template/config/common.yml#L41

I was trying to do something like [this](https://github.com/asyncapi/nodejs-template/blob/fff91aa1f76db4682332f16bb91cdd78fb536ceb/filters/all.js#L173) function does, But the function was too big. So I thought `port` was one of the predefined variables in the spec, but I was wrong. 

I could omit this section to support variables and Just keep it simple. 

What you do think?

Reply:
I'm the author of this beast function 😅 

spec does not specify any convention for variable names. Why can't you take the entire function for replacing variables? In the end you just want to have a "working" URL without variables in it right? We can extract this function from nodejs template to `generator-filters` repo where we have reusable helpers for templates

Reply:
So should I just copy the function for now and you would extract this to a `generator-filter` repo?

Reply:
@derberg .. what you do think about the suggestion I had made?

Reply:
sorry, holidays. The best would be if you could perform the extraction. Not exactly extraction, but just add the same filter to `generator-filters` and then use here (but first make sure it works for you of course. I can later refactor NodeJS template to use this generic filter too

Reply:
@amrutprabhu do you need some support from my side?

Reply:
Hey @derberg ,
So I have updated the code and tested this by right now copying the function `replaceVariablesWithValues` from the `nodejs-template`.

I have also raised a PR in the `generator-filter` repo to have this function as the common function. Here is the PR for it. 
https://github.com/asyncapi/generator-filters/pull/46

Can you have a look at this ?

Reply:
Once the `generator-filters` PR is solved, Do I call the function name directly?

Reply:
@derberg I have fixed this